app/services/achievement-factory.js

```diff
@@ -0,0 +1,291 @@
+import Service from '@ember/service';
+import { inject as service } from '@ember/service';
+import { computed, defineProperty } from '@ember/object';
+import { gte, lte } from '@ember/object/computed';
+import upgrade from 'incremental-empire/utils/upgrade';
+
+export default Service.extend({
+  store: service(),
+  achievementPlan: undefined, // Map that will hold all the achievement definitions
+
+  init() {
+    this._super(...arguments)
+    this.achievementPlan = new Map()
+    this.achievementPlan.set('Reach turn 10', (achievement) => {
+      achievement.setProperties({
+        description: 'You can now assign template points to restart with more population',
+        templatePoint: 1,
+      })
+      defineProperty(achievement, 'condition', gte('game.empire.turn', 10))
+    })
+    this.achievementPlan.set('Reach turn 100', (achievement) => {
+      achievement.setProperties({
+        description: 'A century old empire',
+        templatePoint: 2,
+      })
+      defineProperty(achievement, 'condition', gte('game.empire.turn', 100))
+    })
+    this.achievementPlan.set('Have 10 huts', (achievement) => {
+      achievement.setProperties({
+        description: 'You can now have huts in your templates',
+        templatePoint: 1,
+      })
+      defineProperty(achievement, 'condition', computed('game.empire.buildings.@each.qty', function() {
+        for (let b of achievement.game.empire.buildings) {
+          if (b.name == 'hut') {
+            return b.qty >= 10
+          }
+        }
+        return false
+      }))
```

Comment:
Maybe think of a util to render this easier to write?
```
defineProperty(achievement, '_observedBuilding', empireBuilding('hut'))
setCondition(achievement, gte('_observedBuilding.qty', 10)
```
Make at least one test to verify it's correctly working. Add the test of its effect as well, for good measure.